src/lib.rs

```diff
@@ -74,18 +77,45 @@ mod provider {
 /// println!("The locale is {}", current_locale);
 /// ```
 pub fn get_locale() -> Option<String> {
+    get_locales().next()
+}
+
+/// Returns the preferred locales for the system or application, in descending order of preference.
+///
+/// # Returns
+///
+/// Returns a `Vec` with any number of BCP-47 language tags inside.
+/// If no locale preferences could be obtained, the vec will be empty.
+///
+/// # Example
+///
+/// ```no_run
+/// use sys_locale::get_locales;
+///
+/// let mut  locales = get_locales();
+///
+/// println!("The most preferred locale is {}", locales.next().unwrap_or("en-US".to_string()));
+/// println!("The least preferred locale is {}", locales.last().unwrap_or("en-US".to_string()));
+/// ```
+pub fn get_locales() -> impl Iterator<Item = String> {
     provider::get()
 }
 
 #[cfg(test)]
 mod tests {
-    use super::get_locale;
+    use super::get_locales;
     extern crate std;
 
     #[test]
     fn can_obtain_locale() {
-        let locale = get_locale().expect("locale should be present on most systems");
-        assert!(!locale.is_empty(), "locale string was empty");
-        assert!(!locale.ends_with('\0'), "locale contained trailing NUL");
+        let locales = get_locales();
+        for (i, locale) in locales.enumerate() {
```

Comment:
Strictly speaking it's not an error that the machine has no locales, as we provide for that case in the API - that's why I didn't fix this up earlier. I can add it back though as it's at least *very likely* a developer would have a locale, I guess!

Reply:
FWIW I agree that it is not an error for a machine to have no locales set. This test is just meant to catch _implementation_ errors that might otherwise silently break because all the inner methods don't distinguish between "encountered a fatal error" and "nothing present".